tests/common.go

```diff
@@ -1904,6 +1912,58 @@ func ValidateApplications(contexts []*scheduler.Context) {
 	})
 }
 
+// ValidateApplicationsStartData validates applications and start continous data injection to the same
+func ValidateApplicationsStartData(contexts []*scheduler.Context, appContext context1.Context) (chan string, *errgroup.Group, map[string][]appDriver.ApplicationDriver) {
+
+	// Resetting the global map before starting the new App Validations
+	NamespaceAppWithDataMap = make(map[string][]appDriver.ApplicationDriver)
+
+	log.InfoD("Validate applications")
+	for _, ctx := range contexts {
+		ValidateContext(ctx)
+		appInfo, err := appUtils.ExtractConnectionInfo(ctx)
+		if err != nil {
+			log.InfoD("Some error occurred - [%s]", err)
+		}
+		log.InfoD("App Info - [%+v]", appInfo)
+		if appContext == nil {
+			log.Warnf("App Context is not proper - [%v]", appContext)
+			continue
+		}
+		if appInfo.StartDataSupport {
+			appHandler, _ := appDriver.GetApplicationDriver(
+				appInfo.AppType,
+				appInfo.Hostname,
+				appInfo.User,
+				appInfo.Password,
+				appInfo.Port,
+				appInfo.DBName,
+				appContext,
+				appInfo.NodePort,
+				appInfo.Namespace)
+			log.InfoD("App handler created for [%s]", appInfo.Hostname)
+			NamespaceAppWithDataMap[appInfo.Namespace] = append(NamespaceAppWithDataMap[appInfo.Namespace], appHandler)
+		}
+	}
+
+	controlChannel := make(chan string)
+	errGroup := errgroup.Group{}
```

Comment:
The problem with using errGroup is that it returns only one single error. It returns the error returned by the first goroutine. Are we ok with this?

Reply:
As discussed for now errGroup is fine, we will modify this in future if required.